### DISCUSSION TITLE
Should a component send a console [warn] if slotted light dom isn't what was expected.

### DISCUSSION THREAD

**Discussion by zeroedin** [RedHat-UX/red-hat-design-system#402] [Q&A]
Should a component complain to the console if an expected light dom element is not slotted?

Example:

```
<rh-component-link>
  <div slot="link">Not a hyperlink</div>
</rh-component-link>
```
Console:
`[warn] - [rh-component-link] expects an <a> in <slot name="link">`

Example is purely pseudo-code, not the actual idea of the component in question just a succinct representation

**Comment by heyMP** [comment]
Yeah, I think that's a perfect example for our Logger service from `pfe-core`.  The logger allows use to print warnings to the console when in development mode.  https://github.com/patternfly/patternfly-elements/blob/main/core/pfe-core/controllers/logger.ts

Here is an example of using it similar to your scenario.

```js
import { Logger } from '@patternfly/pfe-core/controllers/logger.js';

export class RhFooter extends LitElement {

  private logger = new Logger(this);

  /**
   * Get any `<ul>`s that are in the designated link slots
   * and syncronously update each list and header if we need to.
   */
  public updateAccessibility(): void {
     ...
        // get the corresponding header that should be the previous sibling
        const header = isHeader(list.previousElementSibling?.tagName ?? '') ? list.previousElementSibling : null;
        if (!header) {
          return this.logger.warn('This links set doesn\'t have a valid header associated with it.');
        } else {
          header.id ||= getRandomId('rh-footer');
          list.setAttribute('aria-labelledby', header.id);
        }
}
```
